packages/block-editor/src/components/block-list/block.native.js

```diff
@@ -248,6 +250,21 @@ function BlockListBlock( {
 		}
 	}, [ selectBlock, clientId, firstToSelectId, isSelected ] );
 
+	const onLayout = useCallback(
+		( { nativeEvent } ) => {
+			const layoutWidth = Math.floor( nativeEvent.layout.width );
+
+			if ( ! blockWidth || ! layoutWidth ) {
+				return;
+			}
+
+			if ( blockWidth !== layoutWidth ) {
+				setBlockWidth( layoutWidth );
```

Comment:
This value is now set in three locations: `BlockList`, `BlockListBlock`, and `BlockListCell`. I draw attention to this to merely verify that this is necessary. 

I believe the first measures as an estimate, the second you added here (what is the reason?), and the third was added for X and Y coordinates for drag-and-drop.

Reply:
I forgot to follow upon the above comment before publishing the review. I modified it slightly for clarity. 

Overarching question: what is the origin for the newly added `onLayout` callback? What problem does it solve? Is the addition a result of moving from `Touchable` to `Pressable`? 

Thanks for helping me understand.

Reply:
This helps immensely. Thank you for expanding. The proposed changes make sense now.